plugins/inputs/hugepages/hugepages.go

```diff
@@ -0,0 +1,291 @@
+	for _, nodeDir := range nodeDirs {
+		if !nodeDir.IsDir() || !strings.HasPrefix(nodeDir.Name(), "node") {
+			continue
+		}
+
+		nodeNumber := strings.TrimPrefix(nodeDir.Name(), "node")
+		_, err := strconv.Atoi(nodeNumber)
+		if err != nil {
+			continue
+		}
+
+		perNodeTags := map[string]string{
+			"node": nodeNumber,
+		}
+		hugepagesPath := filepath.Join(h.numaNodePath, nodeDir.Name(), "hugepages")
+		err = h.gatherFromHugepagePath(hugepagesPath, hugepagesMetricsPerNUMANode, "hugepages_"+perNodeHugepages, perNodeTags, acc)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (h *Hugepages) gatherFromHugepagePath(hugepagesPath string, possibleMetrics []string, measurementName string,
+	tagsToUse map[string]string, acc telegraf.Accumulator) error {
+	// read metrics from: hugepages/hugepages-*/*
+	hugepagesDirs, err := ioutil.ReadDir(hugepagesPath)
+	if err != nil {
+		return err
+	}
+
+	for _, hugepagesDir := range hugepagesDirs {
+		if !hugepagesDir.IsDir() || !strings.HasPrefix(hugepagesDir.Name(), "hugepages-") {
+			continue
+		}
+
+		hugepagesSize := strings.TrimPrefix(strings.TrimSuffix(hugepagesDir.Name(), "kB"), "hugepages-")
+		_, err := strconv.Atoi(hugepagesSize)
+		if err != nil {
+			continue
+		}
+
+		metricsPath := filepath.Join(hugepagesPath, hugepagesDir.Name())
+		metricFiles, err := ioutil.ReadDir(metricsPath)
+		if err != nil {
+			return err
+		}
+
+		metrics := make(map[string]interface{})
+		for _, metricFile := range metricFiles {
+			if mode := metricFile.Mode(); !mode.IsRegular() || !choice.Contains(metricFile.Name(), possibleMetrics) {
+				continue
+			}
+
+			metricBytes, err := ioutil.ReadFile(filepath.Join(metricsPath, metricFile.Name()))
+			if err != nil {
+				return err
+			}
+
+			metricValue, err := strconv.Atoi(string(bytes.TrimSuffix(metricBytes, newlineByte)))
+			if err != nil {
+				return err
+			}
+
+			metrics[metricFile.Name()] = metricValue
+		}
+
+		if len(metrics) > 0 {
+			tags := make(map[string]string)
+			for key, value := range tagsToUse {
+				tags[key] = value
+			}
+			tags["hugepages_size_kb"] = hugepagesSize
+
+			acc.AddFields(measurementName, metrics, tags)
+		}
+	}
+	return nil
+}
+
+// gatherStatsFromMeminfo collects hugepages statistics from meminfo file
+func (h *Hugepages) gatherStatsFromMeminfo(acc telegraf.Accumulator) error {
+	meminfo, err := ioutil.ReadFile(h.meminfoPath)
+	if err != nil {
+		return err
+	}
+
+	metrics := make(map[string]interface{})
+	lines := bytes.Split(meminfo, newlineByte)
+	for _, line := range lines {
+		fields := bytes.Fields(line)
+		if len(fields) < 2 {
+			continue
+		}
+		fieldName := string(bytes.TrimSuffix(fields[0], colonByte))
+		if choice.Contains(fieldName, hugepagesMetricsFromMeminfo) {
+			fieldValue, err := strconv.Atoi(string(fields[1]))
+			if err != nil {
+				return err
+			}
+
+			if bytes.Contains(line, kbPrecisionByte) {
+				fieldName = fieldName + "_kb"
+			}
+			metrics[fieldName] = fieldValue
+		}
```

Comment:
Maybe also invert this
```suggestion
		if !choice.Contains(fieldName, hugepagesMetricsFromMeminfo) {
			continue
		}
		
		fieldValue, err := strconv.Atoi(string(fields[1]))
		if err != nil {
			return err
		}

		if bytes.Contains(line, kbPrecisionByte) {
			fieldName = fieldName + "_kb"
		}
		metrics[fieldName] = fieldValue
```

Reply:
Inverted.